docker/docker-bake-rust-all.hcl

```diff
@@ -2,56 +2,56 @@
 # It provides a high-level mechenanism to build multiple dockerfiles in one shot.
 # Check https://crazymax.dev/docker-allhands2-buildx-bake and https://docs.docker.com/engine/reference/commandline/buildx_bake/#file-definition for an intro.
 
+variable "CI" {
+  # whether this build runs in aptos-labs' CI environment which makes certain assumptions about certain registries being available to push to cache layers.
+  # for local builds we simply default to relying on dockers local caching.
+  default = "false"
+}
 variable "TARGET_CACHE_ID" {}
-
+variable "TARGET_CACHE_TYPE" {
+  // must be "normalized_branch_or_pr" | "git_sha"
+  default = "normalized_branch_or_pr"
+}
 variable "BUILD_DATE" {}
-
-variable "GITHUB_SHA" {}
 // this is the full GIT_SHA - let's use that as primary identifier going forward
-variable "GIT_SHA" {
-  default = "${GITHUB_SHA}"
-}
-// this is the short GIT_SHA (8 chars). Tagging our docker images with that one is kinda deprecated and we might remove this in future.
-variable "GIT_REV" {
-  default = substr("${GIT_SHA}", 0, 8)
-}
+variable "GIT_SHA" {}
 
 variable "LAST_GREEN_COMMIT" {}
 
 variable "GCP_DOCKER_ARTIFACT_REPO" {}
 
 variable "AWS_ECR_ACCOUNT_NUM" {}
 
+variable "TARGET_REGISTRY" {
+  // must be "aws" | "gcp" | "local", informs which docker tags are being generated
+  default = CI == "true" ? "gcp" : "local"
+}
+
 variable "ecr_base" {
   default = "${AWS_ECR_ACCOUNT_NUM}.dkr.ecr.us-west-2.amazonaws.com/aptos"
 }
 
-variable "gh_image_cache" {
-  default = "ghcr.io/aptos-labs/aptos-core"
+variable "normalized_branch_or_pr" {
+  default = regex_replace("${TARGET_CACHE_ID}", "[^a-zA-Z0-9]", "-")
 }
 
-variable "normalized_target_cache_id" {
-  default = regex_replace("${TARGET_CACHE_ID}", "[^a-zA-Z0-9]", "-")
+target "builder" {
+  target     = "builder"
+  dockerfile = "docker/rust-all.Dockerfile"
+  context    = "."
+  cache-from = generate_cache_from("builder")
+  cache-to   = generate_cache_to("builder")
+  tags       = generate_tags("builder")
 }
 
-# images with IMAGE_TARGET=release for rust build
-group "release" {
+group "all" {
   targets = [
     "validator",
     "indexer",
     "node-checker",
-    "safety-rules",
     "tools",
-    "init",
```

Comment:
since we're getting rid of the `init` image, can you also replace this reference? https://github.com/aptos-labs/aptos-core/blob/72f0a1051222867a60c1a057249ac5921d44e699/terraform/helm/genesis/values.yaml#L21

Reply:
done